neqo-http3/src/stream_type_reader.rs

```diff
@@ -108,9 +110,10 @@ impl NewStreamHeadReader {
             reader, stream_id, ..
         } = self
         {
+            let mut buf = [0; MAX_READ_SIZE];
```

Comment:
Does this buffer need to be this hard if it is just reading off the stream head?

Reply:
Not sure I understand what you mean by "this hard"?

Reply:
I mistyped.  I should have said large, but my fingers have a mind of their own sometimes.

The reason I asked is that this type only exists to read at most 16 octets from the stream.  4k is quite a bit more than that.  It's entirely possible that reading more is unproductive.

Reply:
Is there any way to derive this upper limit from the type? (Otherwise I'll make it fixed and explain why in a comment.)

Reply:
A fixed value is fine.  The type doesn't inherently know how much it will need to read, at least not in a way that is easily derived.  I just know that from understanding what it does (which is to read two varints).